tornado/test/httpclient_test.py

```diff
@@ -536,6 +536,15 @@ def test_non_ascii_header(self):
         response.rethrow()
         self.assertEqual(response.headers["Foo"], native_str(u"\u00e9"))
 
+    def test_async_context_manager(self):
+        with self.http_client:
```

Comment:
It's not a good idea to close an object we did not create; this should be `with AsyncHTTPClient(force_instance=True) as client:`

For this reason it's dangerous to use `with AsyncHTTPClient():` because that might be a magically-shared object (should we get rid of that magic? I've filed #2049 to discuss). Maybe we shouldn't implement the context manager protocol on AsyncHTTPClient itself but have another class method like

````
@contextlib.contextmanager
@classmethod
def foo(cls):
    client = cls(force_instance=True)
    try:
        yield client
    finally:
        client.close()
```